channeldb/invoices.go

```diff
@@ -184,6 +186,10 @@ type Invoice struct {
 	// for this invoice can be stored.
 	PaymentRequest []byte
 
+	// PaymentHash is an optional field where a payment hash of this invoice
+	// can be stored.
+	PaymentHash []byte
```

Comment:
This is only _sometimes_ populated when things are read into memory? Not sure if we needed to add this field given that we already have an index over this in the database, and it's available as is within the struct, albeit indirectly.

Reply:
This is only populated in FetchAllInvoices. I myself don't like this solution and only did it on suggestion by other reviewers. The only other way of retrieving the payment hash as far as I can see is by decoding the payment request (which was implemented in an earlier incarnation of this PR).

Reply:
Discussed this further with @joostjager and decided to not populat this field in `FetchAllInvoices()` but rather return to a previously implemented solution, where we select invoices and payment hashes in a differetn call: `FetchAllInvoicesWithPaymentHashes()`.

At one point it would be nice to change ChannelDB to also include the payment hash in Invoice  but it is not possible to do this in all cases without a larger refactor and DB migration in the current implementation.